sources/packages/api/src/services/atbc/atbc.service.ts

```diff
@@ -0,0 +1,180 @@
+import { Injectable } from "@nestjs/common";
+import { ConfigService } from "../config/config.service";
+import axios from "axios";
+import { LoggerService } from "../../logger/logger.service";
+import { InjectLogger } from "../../common";
+import {
+  ATBCHeader,
+  ATBCAuthTokenResponse,
+} from "../auth/token-cache.service.models";
+import {
+  ATBCIntegrationConfig,
+  ATBCCreateClientResponse,
+  ATBCCreateClientPayload,
+  ATBCPDCheckerResponse,
+  ATBCPDCheckerPayload,
+} from "../../types";
+import { Student } from "../../database/entities";
+import { StudentService } from "../../services";
+
+@Injectable()
+export class ATBCService {
+  constructor(
+    private readonly configService: ConfigService,
+    private readonly studentService: StudentService,
+  ) {
+    this.logger.log("[Created]");
+  }
+  /**
+   * Allow a singleton access to the ATBCService for areas
+   * of the application that are not part of the dependency
+   * injection framework.
+   */
+  public static get shared(): ATBCService {
+    return ATBCService.shared;
+  }
+
+  get config(): ATBCIntegrationConfig {
+    return this.configService.getConfig().ATBCIntegration;
+  }
+  /**
+   * Executes the token request
+   * @returns bearer header.
+   */
+  private async getConfig(): Promise<ATBCHeader> {
+    const token = await this.getAuthToken();
+    return {
+      headers: { Authorization: `Bearer ${token}` },
+      // (NOTE: this will disable client verification)
+      // TODO: add certificate for PROD
+      httpsAgent: new (require("https").Agent)({
```

Comment:
This is really important and we cannot forget about it. Please create a tech debit for this.